fuel-benches/benches/set/mem.rs

```diff
@@ -0,0 +1,131 @@
+use super::run_group;
+
+use criterion::Criterion;
+use fuel_core_benches::*;
+
+pub fn run(c: &mut Criterion) {
+    let mut group = c.benchmark_group("mem");
+
+    let linear = vec![1, 10, 100, 1_000, 10_000, 100_000];
+
+    for i in &linear {
+        run_group(
+            &mut group,
+            format!("aloc ({})", i),
+            VmBench::new(Opcode::ALOC(0x10))
+                .with_prepare_script(vec![Opcode::MOVI(0x10, *i)]),
+        );
+    }
+
+    for i in &linear {
+        run_group(
+            &mut group,
+            format!("cfei ({})", i),
+            VmBench::new(Opcode::CFEI(*i)),
+        );
+    }
+
+    for i in &linear {
+        run_group(
+            &mut group,
+            format!("cfsi ({})", i),
+            VmBench::new(Opcode::CFSI(*i)).with_prepare_script(vec![Opcode::CFEI(*i)]),
+        );
+    }
+
+    run_group(
+        &mut group,
+        "lb",
+        VmBench::new(Opcode::LB(0x10, REG_ONE, 10)),
+    );
+
+    run_group(
+        &mut group,
+        "lw",
+        VmBench::new(Opcode::LW(0x10, REG_ONE, 10)),
+    );
+
+    for i in &linear {
+        run_group(
+            &mut group,
+            format!("mcl ({})", i),
+            VmBench::new(Opcode::MCL(0x10, 0x11)).with_prepare_script(vec![
+                Opcode::MOVI(0x11, *i),
+                Opcode::ALOC(0x11),
```

Comment:
Could be nice to have comments explaining why each script was chosen for each benchmark for those consulting benchmarks and to keep every up to speed with logic used for benchmarking

Reply:
Cool, will add some docs and simplify the code a bit